visualizations/frontend/filters/v2/TDatePicker.vue

```diff
@@ -0,0 +1,282 @@
+<template>
+    <div class="relative flex t-date-picker">
+        <!-- Date Picker -->
+        <base-date-picker
+            read-only
+            show-clear-button
+            confirm
+            confirm-text="Apply"
+            v-model="date"
+            ref="datePicker"
+            :range="mode === 'range'"
+            :format="dateFormat"
+            :range-separator="seperator"
+            @confirm="handleConfirm"
+            @pick="setCustomPreset"
+            @open="fetchLayerData"
+        >
+            <div class="presets-dropown" slot="sidebar">
+                <div
+                    v-for="preset in presets[mode]"
+                    :key="preset.key"
+                    :style="{ color: preset.key === selectedPreset.key ? '#1284e7' : '#73879c' }"
+                    class="cursor-pointer"
+                    @click="handlePreset(preset)"
+                >
+                    {{ preset.label }}
+                </div>
```

Comment:
could we make these buttons or other focusable elements? I think it is better practice for clickable elements.